plugins/catalog-react/src/components/EntityRefLink/FetchedEntityRefLinks.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Entity,
+  CompoundEntityRef,
+  parseEntityRef,
+} from '@backstage/catalog-model';
+import React from 'react';
+import { EntityRefLink } from './EntityRefLink';
+import { ErrorPanel, LinkProps, Progress } from '@backstage/core-components';
+import useAsync from 'react-use/lib/useAsync';
+import { catalogApiRef } from '../../api';
+import { useApi } from '@backstage/core-plugin-api';
+
+/**
+ * Props for {@link FetchedEntityRefLinks}.
+ *
+ * @public
+ */
+export type FetchedEntityRefLinksProps<
+  TRef extends string | CompoundEntityRef | Entity,
+> = {
+  defaultKind?: string;
+  entityRefs: TRef[];
+  getTitle(entity: Entity): string | undefined;
+} & Omit<LinkProps, 'to'>;
+
+/**
+ * Shows a list of clickable links to entities with auto-fetching of entities
+ * for customising a displayed text via title attribute.
+ *
+ * @public
+ */
+export function FetchedEntityRefLinks<
+  TRef extends string | CompoundEntityRef | Entity,
+>(props: FetchedEntityRefLinksProps<TRef>) {
+  const { entityRefs, defaultKind, getTitle, ...linkProps } = props;
+
+  const catalogApi = useApi(catalogApiRef);
+
+  const {
+    value: entities = new Array<Entity>(),
+    loading,
+    error,
+  } = useAsync(async () => {
+    const refs = entityRefs.reduce((acc, current) => {
+      return 'metadata' in current ? acc : [...acc, parseEntityRef(current)];
```

Comment:
We're throwing away any full entities here right? Would need to merge them back together into the array afterwards. Being a bit pedantic I think it'd be best if the order is maintained as well, so best cover that bit with a test to make sure it's solid.

Reply:
I'm assuming that user can work only with the same type in array, not a mix of ids, refs and entities.
If it's an array of entities it just used as is

return refs
      ? (
          await catalogApi.getEntities({
            filter: refs.map(ref => ({
              kind: ref.kind,
              'metadata.namespace': ref.namespace,
              'metadata.name': ref.name,
            })),
          })
        ).items
      : (entityRefs as Array<Entity>);

Reply:
Alright let's either have that reflected in types or throw an error if a mix is provided. Right now there's a bit of a risk of confusion I'd say. Then we can slap a `// TODO` on it to potentially handle it later